File: packages/amplify_core/lib/src/types/auth/sign_up/auth_sign_up_step.dart
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

/// {@category Auth}
/// {@template amplify_core.auth.auth_sign_up_step}
/// The current step in the sign up flow.
/// {@endtemplate}
enum AuthSignUpStep {
  /// The user is successfully registered but requires an additional
  /// confirmation before the sign up is considered complete.
  confirmSignUp,

  /// The user is successfully registered and sign up is complete.
  done,
}

File: packages/amplify_datastore/darwin/amplify_datastore/Sources/amplify_datastore/AmplifyInternal/Categories/Auth/Models/AuthSignUpStep.swift
//
// Copyright Amazon.com Inc. or its affiliates.
// All Rights Reserved.
//
// SPDX-License-Identifier: Apache-2.0
//

public typealias UserId = String

/// SignUp step to be followed.
public enum AuthSignUpStep {

    /// Need to confirm the user
    case confirmUser(
        AuthCodeDeliveryDetails? = nil,
        AdditionalInfo? = nil,
        UserId? = nil)

    /// Sign up is complete
    case done
}
